### PR TITLE
chore(angular): split module-federation tests into their own project

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

Module Federation tests for angular take a significant amount of time on their own.
Having them as part of a project with other test suites increases the overall test time.

## Expected Behavior

Split the Module Federation tests into their own project to allow it to be run in parallel

Doing so shows a ~50% decrease in time to run angular e2es.
**Before**
image

**After**
image

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by jaysoo — APPROVED** [review]

**Review by leosvelperez — CHANGES_REQUESTED** [review]
The new e2e project must be added to the nightly CI pipelines:

- `.github/workflows/e2e-matrix.yml`
- `.github/workflows/e2e-windows.yml`

File: .github/workflows/e2e-matrix.yml [hunk 1]
```diff
@@ -214,6 +217,8 @@ jobs:
             project: e2e-angular-core
           - node_version: 16
             project: e2e-angular-extensions
+          - node_version: 16
+            project: e2e-angular-module-federation
```
Comment (leosvelperez) [inline .github/workflows/e2e-matrix.yml:221]: It should also be added for `node_version: 20`

File: .github/workflows/e2e-windows.yml [hunk 2]
```diff
@@ -159,6 +162,8 @@ jobs:
             project: e2e-angular-core
           - node_version: 16
             project: e2e-angular-extensions
+          - node_version: 16
+            project: e2e-angular-module-federation
```
Comment (leosvelperez) [inline .github/workflows/e2e-windows.yml:166]: It should also be added for `node_version: 20`

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-matrix.yml (hunk 1 above)
Comment (Coly010) [reply .github/workflows/e2e-matrix.yml:221]: it is? right? 
image

File: .github/workflows/e2e-windows.yml (hunk 2 above)
Comment (Coly010) [reply .github/workflows/e2e-windows.yml:166]: it is? right? 
image

File: .github/workflows/e2e-matrix.yml (hunk 1 above)
Comment (leosvelperez) [reply .github/workflows/e2e-matrix.yml:221]: That's correctly added to the matrix, but we exclude the other angular e2e projects from running with node 16 and 20. The new one need to match that.

**Review by leosvelperez — APPROVED** [review]

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**